src/pool/srv_iv.c

```diff
@@ -248,6 +280,78 @@ pool_iv_update(void *ns, struct pool_iv_entry *pool_iv,
 	return rc;
 }
 
+int
+pool_iv_invalidate(void *ns, unsigned int shortcut, unsigned int sync_mode)
+{
+	struct ds_iv_key	key = { 0 };
+	int			rc;
+
+	key.class_id = IV_POOL_MAP;
+	rc = ds_iv_invalidate(ns, &key, shortcut, sync_mode, 0);
+	if (rc)
+		D_ERROR("iv invalidate failed %d\n", rc);
+
+	return rc;
+}
+
+/* ULT to refresh pool map version */
+void
+ds_pool_iv_refresh_ult(void *arg)
+{
+	struct pool_iv_refresh_ult_arg *iv_arg = arg;
+	d_rank_t	rank;
+	struct ds_pool *pool;
+	int rc = 0;
+
+	/* Pool IV fetch should only be done in xstream 0 */
+	D_ASSERT(dss_get_module_info()->dmi_xs_id == 0);
+	pool = ds_pool_lookup(iv_arg->iua_pool_uuid);
+	if (pool == NULL) {
+		rc = -DER_NONEXIST;
+		goto out;
+	}
+
+	rc = crt_group_rank(pool->sp_group, &rank);
+	if (rc)
+		goto out;
+
+	if (rank == pool->sp_iv_ns->iv_master_rank) {
+		D_WARN("try to refresh pool map on pool leader\n");
+		goto out;
+	}
+
+	/* If there are already refresh going on, let's wait
+	 * until the refresh is done.
+	 */
+	ABT_mutex_lock(pool->sp_iv_refresh_lock);
+	if (pool->sp_map_version >= iv_arg->iua_pool_version &&
+	    pool->sp_map != NULL &&
+	    !DAOS_FAIL_CHECK(DAOS_FORCE_REFRESH_POOL_MAP)) {
+		D_DEBUG(DB_TRACE, "current pool version %u >= %u\n",
+			pool_map_get_version(pool->sp_map),
+			iv_arg->iua_pool_version);
+		goto unlock;
+	}
+
+	/* Invalidate the local pool IV cache, then call pool_iv_fetch to
+	 * refresh local pool IV cache, which will update the local pool map
+	 * see pool_iv_ent_refresh()
+	 */
+	rc = pool_iv_invalidate(pool->sp_iv_ns, CRT_IV_SHORTCUT_NONE,
+				CRT_IV_SYNC_NONE);
+	if (rc)
+		goto unlock;
+
+	rc = pool_iv_fetch(pool->sp_iv_ns, NULL);
```

Comment:
no any version parameter, did not get how to check if the pool map version is that fetch wanted version?
for example in node A found pool map stale, then in ds_pool_iv_refresh_ult() it do "invalidate + fetch", however if the fetched pool map (from parent) is still old version?

Reply:
Hmm, the idea is to fetch the updated pool map from the leader and client will resend the request, if it still stale, it will fetch again. 
Hmm, maybe we can add version here for optimizations, the issue is that the daos IV interface does not provide extra parameter for fetch, except one key, which is pool uuid in this case. I will think about it. Thanks.